framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -487,95 +384,72 @@ private static Receiver getReceiverField(
         return new FieldAccess(locationOfField, fieldType, fieldElem);
     }
 
-    private static boolean isParameter(String s, FlowExpressionContext context) {
-        if (context.parsingMember) {
-            return false;
-        }
-        Matcher parameterMatcher = PARAMETER_PATTERN.matcher(s);
-        return parameterMatcher.matches();
-    }
-
     private static Receiver parseParameter(String s, FlowExpressionContext context)
             throws FlowExpressionParseException {
-        Matcher parameterMatcher = PARAMETER_PATTERN.matcher(s);
-        if (!parameterMatcher.matches()) {
-            return null;
-        }
         if (context.arguments == null) {
             throw constructParserException(s, "no parameter found");
         }
-        int idx = -1;
-        try {
-            idx = Integer.parseInt(parameterMatcher.group(1));
-        } catch (NumberFormatException e) {
-            // cannot occur by the way the pattern is defined (matches only numbers)
-            assert false;
-        }
+        int idx = Integer.parseInt(s.substring(7));
```

Comment:
Where does the number 7 come from here?

Reply:
It's because `"_param_".length()  = 7`, and the substring that start after `"_param_"` is the parameter number. Parameters are now recognized as `"_param_"` instead of `"#"`.

Reply:
Okay, that makes sense. Can you add a comment that explains the magic number here?